controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -405,22 +515,21 @@ public String getScopeName() {
     }
 
     private CompletableFuture<TxnStatus> checkTransactionStatus(final int epoch, final UUID txId) {
-        final CompletableFuture<TxnStatus> activeTx = getActiveTx(epoch, txId).handle((ok, ex) -> {
+
+        return verifyLegalState(() -> getActiveTx(epoch, txId).handle((ok, ex) -> {
```

Comment:
Same here, why is `verifyLegalState` check needed for read operation like `checkTransactionStatus`?